From d53773e3da3cd0c0e19d0e382c4461062955f885 Mon Sep 17 00:00:00 2001
From: David Ward <david.ward@ll.mit.edu>
Date: Mon, 2 Apr 2012 23:45:35 -0400
Subject: [PATCH] service: schedule JOB_RESTART from SERVICE_AUTO_RESTART
 state

Fixes: https://bugs.freedesktop.org/show_bug.cgi?id=45511
(cherry picked from commit 48bb58769a00e8e50a617ebbfff84599a0350fa5)

Conflicts:
	src/service.c
---
 src/service.c |    9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/src/service.c b/src/service.c
index 97a4a45..3ed79b0 100644
--- a/src/service.c
+++ b/src/service.c
@@ -2125,9 +2125,12 @@ static void service_enter_restart(Service *s) {
                         goto fail;
         }
 
-        service_enter_dead(s, true, false);
-
-        if ((r = manager_add_job(UNIT(s)->manager, JOB_START, UNIT(s), JOB_FAIL, false, &error, NULL)) < 0)
+        /* Any units that are bound to this service must also be
+         * restarted. We use JOB_RESTART (instead of the more obvious
+         * JOB_START) here so that those dependency jobs will be added
+         * as well. */
+        r = manager_add_job(UNIT(s)->manager, JOB_RESTART, UNIT(s), JOB_FAIL, false, &error, NULL);
+        if (r < 0)
                 goto fail;
 
         log_debug("%s scheduled restart job.", UNIT(s)->id);
